metaflow/plugins/airflow/airflow_utils.py

```diff
@@ -0,0 +1,410 @@
+    secrets = [
+        Secret("env", secret, secret) for secret in operator_args.get("secrets", [])
+    ]
+    args = operator_args
+    args.update(
+        {
+            "secrets": secrets,
+            # Question for (savin):
+            # Default timeout in airflow is 120. I can remove `startup_timeout_seconds` for now. how should we expose it to the user?
+        }
+    )
+    # Below cannot be passed in dictionary form. After trying a few times it didin't work.
+    additional_env_vars = [
+        client.V1EnvVar(
+            name=k,
+            value_from=client.V1EnvVarSource(
+                field_ref=client.V1ObjectFieldSelector(field_path=str(v))
+            ),
+        )
+        for k, v in {
+            "METAFLOW_KUBERNETES_POD_NAMESPACE": "metadata.namespace",
+            "METAFLOW_KUBERNETES_POD_NAME": "metadata.name",
+            "METAFLOW_KUBERNETES_POD_ID": "metadata.uid",
+            "METAFLOW_KUBERNETES_SERVICE_ACCOUNT_NAME": "spec.serviceAccountName",
+        }.items()
+    ]
+    args["env_vars"] = [
+        client.V1EnvVar(name=x["name"], value=x["value"]) for x in args["env_vars"]
+    ] + additional_env_vars
+
+    # We need to explicitly parse resources to k8s.V1ResourceRequirements otherwise airflow tries
+    # to parse dictionaries to `airflow.providers.cncf.kubernetes.backcompat.pod.Resources` object via
+    # `airflow.providers.cncf.kubernetes.backcompat.backward_compat_converts.convert_resources` function
+    resources = args.get("resources")
+    args["resources"] = client.V1ResourceRequirements(
+        requests=resources["requests"],
+    )
+    if operator_args.get("execution_timeout", None):
+        args["execution_timeout"] = timedelta(
+            **operator_args.get(
+                "execution_timeout",
+            )
+        )
+    if operator_args.get("retry_delay", None):
+        args["retry_delay"] = timedelta(**operator_args.get("retry_delay"))
+    return args
+
+
+def _parse_sensor_args(name, kwargs):
+    if name == SensorNames.EXTERNAL_TASK_SENSOR:
+        if "execution_delta" in kwargs:
+            if type(kwargs["execution_delta"]) == dict:
+                kwargs["execution_delta"] = timedelta(**kwargs["execution_delta"])
+            else:
+                del kwargs["execution_delta"]
+    return kwargs
+
+
+def _get_sensor(name):
+    if name == SensorNames.EXTERNAL_TASK_SENSOR:
+        # ExternalTaskSensors uses an execution_date of a dag to
+        # determine the appropriate DAG.
+        # This is set to the exact date the current dag gets executed on.
+        # For example if "DagA" (Upstream DAG) got scheduled at
+        # 12 Jan 4:00 PM PDT then "DagB"(current DAG)'s task sensor will try to
+        # look for a "DagA" that got executed at 12 Jan 4:00 PM PDT **exactly**.
+        # They also support a `execution_timeout` argument to
+        from airflow.sensors.external_task_sensor import ExternalTaskSensor
```

Comment:
Currently, it seems that if I have to define a new sensor, I need to make changes in many different places. Can you look into consolidating all this logic and mappings in a single spot?

Reply:
I can do it more dynamically by importing these sensors via an `importlib` and removing the SensorName stuff altogether. I felt that was overkill so I kept it this way. We can modify this in the near future but for now, we can keep it like this as the integration is not catering to too many sensors.